### PR TITLE
fix: Promotion to main - spring boot 3

### PR DESCRIPTION
**Pull request by and-mora**
#### How Has This Been Tested?

- Pre-Deploy Test
  - [ ] Unit
  - [ ] Integration (Narrow)
- Post-Deploy Test
  - [ ] Isolated Microservice
  - [ ] Broader Integration
  - [ ] Acceptance
  - [ ] Performance & Load

#### Types of changes

- [ ] PATCH - Bug fix (backwards compatible bug fixes)
- [ ] MINOR - New feature (add functionality in a backwards compatible manner)
- [ ] MAJOR - Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by TommasoLencioni — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -17,13 +17,13 @@ jobs:
...
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           persist-credentials: false
           fetch-depth: 0
 
       - name: Release
-        uses: cycjimmy/semantic-release-action@v2
+        uses: cycjimmy/semantic-release-action@v3
```
Comment (GiovanniMancini) [inline .github/workflows/release.yml:26]: @and-mora Is `v3` the last tagged version on action repo?

File: .releaserc.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "branches": [
+    "main",
+    "master"
```
Comment (GiovanniMancini) [inline .releaserc.json:4]: @and-mora Do we have a `master` branch?

File: Dockerfile [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-FROM amazoncorretto:11 as runtime
+FROM amazoncorretto:17.0.6-al2023 as runtime
 
 VOLUME /tmp
 WORKDIR /app
 
 COPY --from=buildtime /build/target/*.jar /app/app.jar
 # The agent is enabled at runtime via JAVA_TOOL_OPTIONS.
-ADD https://github.com/microsoft/ApplicationInsights-Java/releases/download/3.4.7/applicationinsights-agent-3.4.7.jar /app/applicationinsights-agent.jar
+ADD https://github.com/microsoft/ApplicationInsights-Java/releases/download/3.4.11/applicationinsights-agent-3.4.11.jar /app/applicationinsights-agent.jar
```
Comment (GiovanniMancini) [inline Dockerfile:15]: @and-mora Microsoft released `3.4.13` version of App Insight Agent. Do you think it's still immature?

Comment (and-mora) [reply Dockerfile:15]: we haven't tested it yet. We can bump it in the next release and test in dev/uat env.

File: .releaserc.json (hunk 2 above)
Comment (and-mora) [reply .releaserc.json:4]: nope, i think it's derived from a copy&paste of the `.releaserc.json` which fits both branch naming

**Pull request merged**